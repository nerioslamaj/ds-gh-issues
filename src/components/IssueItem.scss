@import '../index.scss';

.Issue-item-wrapper {
    background-color: $dark-bcg-2;
    padding: 16px 24px;
    border-bottom: 1px solid $border-col;

    .Issue-item-title {
        font-size: 1rem;
        font-weight: 600;
        margin-bottom: 10px;
        line-height: 1.5;

        .Issue-item-label {
            font-size: 0.75rem;
            display: inline-block;
            font-weight: 600;
            padding: 3px 7px 4px;
            line-height: 1.2rem;
            margin-left: 8px;
            border-radius: 100px;
        }
    }

    .Issue-item-subtitle {
        color: $light-txt-2;
        font-size: 0.75rem;
    }
}