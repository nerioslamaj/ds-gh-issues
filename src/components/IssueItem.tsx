import React from 'react';
import './IssueItem.scss';

type IssueItemProp = {
  title: String,
  author: String,
  createdAt: String,
  number: Number,
  labels: Array<any>
};

class App extends React.Component< IssueItemProp > {
  constructor(props: IssueItemProp){
      super(props);
      this.state = {};
  }

  render() {
    return (
      <div className="List-item-wrapper">
        { this.props.title }
        { this.props.author }
        { this.props.createdAt }
        { this.props.title }
      </div>
    );
  }
}

export default App;
