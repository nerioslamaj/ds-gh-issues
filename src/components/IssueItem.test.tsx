import React from 'react';
import IssueItem from './IssueItem';
import TestRenderer from 'react-test-renderer';

it('renders correctly', () => {

  const issue = {
    number: 1,
    title: 'Item Title',
    created_at: '2020-08-09T21:38:06Z',
    labels: ['label 1', 'label 2'],
    html_url: 'https://example.url',
    user: {
      login: 'NeriosLamaj'
    }
  }

  const tree = TestRenderer
    .create(<IssueItem 
      key={issue.number}
      title={issue.title}
      author={issue.user.login}
      number={issue.number}
      createdAt={issue.created_at}
      labels={issue.labels}
      url={issue.html_url}></IssueItem>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
