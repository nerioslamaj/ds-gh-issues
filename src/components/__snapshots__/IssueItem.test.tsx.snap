// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Issue-item-wrapper"
>
  <h3
    className="Issue-item-header"
  >
    <a
      className="Issue-item-title"
      href="https://example.url"
      rel="noopener noreferrer"
      target="_blank"
    >
      Item Title
    </a>
    <a
      className="Issue-item-label"
      style={
        Object {
          "backgroundColor": "#undefined",
        }
      }
    />
    <a
      className="Issue-item-label"
      style={
        Object {
          "backgroundColor": "#undefined",
        }
      }
    />
  </h3>
  <p
    className="Issue-item-subtitle"
  >
    <a
      href="https://github.com/facebook/create-react-app/issues/1"
      rel="noopener noreferrer"
      target="_blank"
    >
      #
      1
    </a>
     opened on 
    09.08.2020
     by 
    <a
      href="https://github.com/NeriosLamaj"
      rel="noopener noreferrer"
      target="_blank"
    >
      NeriosLamaj
    </a>
  </p>
</div>
`;
